server/src/main/java/io/druid/metadata/IndexerSQLMetadataStorageCoordinator.java

```diff
@@ -88,11 +88,13 @@ public IndexerSQLMetadataStorageCoordinator(
             final ResultIterator<byte[]> dbSegments =
                 handle.createQuery(
                     String.format(
-                        "SELECT payload FROM %s WHERE used = true AND dataSource = :dataSource",
+                        "SELECT payload FROM %s WHERE used = true AND dataSource = :dataSource AND start >= :start and \"end\" <= :end  AND used = true",
```

Comment:
I can see arguments for both

`getUnusedSegmentsForInterval` is typically used to get segments that can safely be deleted. If the goal is to apply some kind of retention policy, someone would probably not want to include segments that simply overlap, since they may still contain more recent data beyond the end interval.

`getUsedSegmentsForInterval` is used as part of the merge task to ensure we get all the segments that might contain data for a given interval. If we do not include overlapping segments we are not accounting for all the segments and the resulting merged segment would not have all the data.

The best solution is probably to have separate methods to indicate the intent, e.g `getUnusedSegmentsOverlappingInterval` and `getUnusedSegmentsContainedInInterval`
and the same for used segments